libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewApi.kt

```diff
@@ -1,108 +1,145 @@
+     * Route Preview state.
      *
-     * @param routes list of [NavigationRoute]
+     * Fails with an error when either Destination or Preview Routes has not been set.
      */
-    fun setPreviewRoutes(routes: List<NavigationRoute>) {
-        store.dispatch(RoutePreviewAction.Ready(routes))
-    }
+    abstract fun startRoutePreview(): Expected<NavigationViewApiError, Unit>
 
     /**
-     * Sets the [routes] passed to [MapboxNavigation]. The action triggers [RouteProgress],
-     * [BannerInstructions] and [VoiceInstructions] updates if the [TripSessionState] is
-     * [TripSessionState.STARTED].
+     * Sets a preview [routes] and request Request [NavigationView] to enter Route Preview state.
+     * Last [DirectionsWaypoint] location will be used as the destination.
+     *
+     * [NavigationViewListener.onDestinationChanged] will be called once new destination is set
+     * and [NavigationViewListener.onRoutePreview] will be called once [NavigationView] enters
+     * Route Preview state.
      *
-     * @param routes list of [NavigationRoute]
+     * Fails with an error when [routes] is an empty list.
      */
-    fun setRoutes(routes: List<NavigationRoute>) {
-        store.dispatch(RoutesAction.SetRoutes(routes))
-    }
+    abstract fun startRoutePreview(
+        routes: List<NavigationRoute>
+    ): Expected<NavigationViewApiError, Unit>
 
     /**
-     * Sets a destination to [NavigationView].
-     * @property point the destination location
+     * Request [NavigationView] to enter Active Navigation state.
+     *
+     * [NavigationViewListener.onActiveNavigation] will be called once [NavigationView] enters
+     * Route Preview state.
+     *
+     * Fails with an error when either Destination or Preview Routes has not been set.
+     */
+    abstract fun startActiveGuidance(): Expected<NavigationViewApiError, Unit>
+
+    /**
+     * Sets [routes] and request [NavigationView] to enter Active Navigation state.
+     * Last [DirectionsWaypoint] location will be used as the destination.
+     *
+     * [NavigationViewListener.onDestinationChanged] will be called once new destination is set
+     * and [NavigationViewListener.onActiveNavigation] will be called once [NavigationView] enters
+     * Active Navigation state.
+     *
+     * Fails with an error when [routes] is an empty list.
      */
-    fun setDestination(point: Point) {
-        store.dispatch(DestinationAction.SetDestination(Destination(point)))
-    }
+    abstract fun startActiveGuidance(
+        routes: List<NavigationRoute>
+    ): Expected<NavigationViewApiError, Unit>
 
     /**
-     * Enables trip session based on real gps updates.
+     * Request [NavigationView] to enter Arrival state.
+     *
+     * [NavigationViewListener.onArrival] will be called once [NavigationView] enters
+     * Arrival state.
+     *
+     * Fails with an error when either Destination or Routes has not been set.
```

Comment:
As with `startActiveGuidance` this limitation also doesn't let replace `SharedApp.store.dispatch(NavigationStateAction.Update(NavigationState.Arrival))` with `startArrival` in 1TAP. The use case where this doesn't work for us is showing arrival feedback for the trip that happened a while ago. We save metadata for the trip that is necessary for reporting the feedback later, but we can't afford saving the whole routes.